### DISCUSSION TITLE
Add a magnifying glass to search by habit name

### DISCUSSION THREAD

**Discussion by KeronCyst** [iSoron/uhabits#676] [Feature Requests] [answered]
I have 40 habits set, so it can be difficult to find one that I'm looking for. Being able to run a search to quickly locate a habit in the overview would be super-helpful!

**Comment by ljl-covid** [comment]
I second this... I actually just came here to file a feature request about this, but looks like I've been preceded by six days.

The functionality may either filter through habits by name, search for a specific habit, or scroll to the habit searched for and highlight it; I don't know what would be better UX... I suspect filtering or scrolling+highlighting would work better for me personally, but ultimately, just like @KeronCyst, I have an unwieldy number of habits, and I've spent enough time staring at them that I could probably have written this feature request a thousand times in that time. Maybe it's not a typical use case, but hey, there's at least two of us!

**Comment by hiqua** [comment]
Why not I suppose, but I guess most users have something like 10 habits, and users with 40+ habits are outliers, so I don't think it will benefit many users.

Even for users with a lot of habits, what about better sorting / colors to quickly find the habit you need?

Another point: I'd have to time myself, but I'd guess finding a habit myself by scrolling would be faster than typing on a smartphone, and less error-prone obviously.

Actually I can say with confidence that with my ~30 habits, I wouldn't use a search function. Maybe with 60 habits it would start making sense (for me).

But if you want to open a PR I think it might be merged, since it should not conflict with other stuff.

**Reply by pandaseal** [reply]
> users with 40+ habits are outliers

Maybe this needs to be researched a bit to be certain. Personally I have 100+ habits and it seems like I'm not the only one having lots of habits.

**Comment by iSoron** [comment]
@KeronCyst @ljl-covid I'm not sure I understand the workflow here. After searching for the habit, what operation do you do next?  Do you (i) search for the habit, (ii) add a checkmark, (iii) search for another habit, (iv) add another checkmark? As @hiqua mentioned, I'm not sure a search function would actually save any time in this workflow, due to all the typing involved. Perhaps collapsible groups of habits (as suggested in #495) would work better.

If anyone thinks this could work and is interested in implementing this, you are very welcome, but I would like to see a mockup of the proposed user interface first.

**Comment by ljl-covid** [comment]
@iSoron yes, pretty much. I have many habits, admittedly likely many more than the average user (but do we actually know how the average user uses this apps? I don't even use it for things that you'd properly call "habits", for the most part), and I think that workflow would be fast for me.

I currently use color codes but that's not enough, and I keep staring anyway. Typing a couple of letters in a search filter would likely reduce the number of shown habits to just a couple, I don't feel like there would be *a lot* of typing involved.

Collapsible groups may work better for me than the current color coding, and I'd probably use those even with a search feature present, but I suspect I'd probably use the search feature the most.

I understand this may not reflect how others use the apps or the way you envision it, though. That's fine.

**Comment by KeronCyst** [comment]
@iSoron it's not usually to checkmark; I sometimes edit the frequency of a somewhat-new habit as I get in the groove of what I'm trying to do with it over weeks of experience with it (or I just messed up the initial frequency accidentally), and then I try to look through the list by frequency. So, sorting by frequency would be nice. I will try colors; I had forgotten about those. Collapsible groups sound like a great alternative, too!

**Reply by dtgoitia** [reply]
> Typing a couple of letters in a search filter would likely reduce the number of shown habits to just a couple

That's exactly what I had in mind.

Colour coding is not enough for me either. Grouping habits would force me to build a hierarchy and remember it. Searching would only require me to remember the name of the habit.

Use case: track injuries or sickness symptoms over time. They are not habits, but still the time analysis tools of the app are very handy for this use case.

Any chance to reconsider this feature request?

**Reply by pandaseal** [reply]
> Use case: track injuries or sickness symptoms over time. They are not habits, but still the time analysis tools of the app are very handy for this use case.

Same here! I have over 120 data points (habits) that I track with various frequency (most are archived most of the time) and I would love to just be able to search for a key word to narrow down the list of habits, or to find habits that I don't remember the exact name of.

My solution now is to prefix all trackers/habits with like "Symtom: <habit name>" or "Event: <habit name>" and make use sorting by name to find them, but this is not efficient and it requires me to remember what name the habit that I'm searching for has.

**Comment by alittlebitofit** [comment]
Is there an update to this?
Is this being implemented?

Search feature would be used by those who have lots of habits.
Those who don't want to use it, won't use it.

So it doesn't disturb anybody's workflow.

I have lots of habits that I have archived and update them like once a week. I always forget where in the list a particular habit is.
And it takes a LOT of time to find it, and by that time I am already annoyed. This happened too many times so I'd like to ease my workflow with a search feature.

PS:
1) How do you know an average user has fewer habits than most?
2) Why does clicking on "Open New Feature Request" take me to Discussions tab instead of letting me submit the report?

**Reply by hiqua** [reply]
> Is there an update to this?

All the dev is in the open, so there's not more than you can see. Feel free to follow the steps in https://github.com/iSoron/uhabits/discussions/676#discussioncomment-245647 to suggest an implementation.

> Why does clicking on "Open New Feature Request" take me to Discussions tab instead of letting me submit the report?

The feature requests are kept in the Discussions tab rather than the issues, see: https://github.com/iSoron/uhabits/issues/684#issue-775136496

**Comment by mohanrajkumarplus** [comment]
Hi Team , any update to this feature , I have more than 200+ habits. i know it sounds crazy , but suppose let's take daily walking as an habit , so I have step based sub habits to it . so a search functionality would be a great addition. category would be helpful , still the scrolling would be happening if there are more groups.

**Comment by Akhillalpp** [comment]
I track a lot of habits, so it can be tedious to scroll through the list just to find one to log. A search bar would make logging much faster.

It would be even more useful if search also worked for archived habits. I keep event-based habits (like migraine attacks) archived and only unarchive them when I need to log an occurrence. Searching archived habits would make this workflow much more convenient.

**Comment by iSoron** [comment, marked as answer]
This feature has been implemented in 4a930f5e40292866c683fb43b87efe241532b114 and will be available in Loop 2.4.0. The search function is within the filter menu, below sorting.
